### PR TITLE
Crew monitor revisit

### PR DESCRIPTION
**Pull request by chromiumboy**
## About the PR

This PR retouches the UI for the crew monitoring console

## Why / Balance

Hopefully it looks better now, and with better performance

## Technical details

At its heart its the same as before. The biggest changes are in NavMapControl.cs

After reviews are complete I'll file scope the name spaces; it's been left out for now to make reviewing easier

## Media

[Video of the new UI in action](https://youtu.be/XnF1mc8vd3Q)

Updated UI:

crew_monitoring_update

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

SuitSensorStatus has been changed, it now takes four arguments: the user's name (string), job title (string), job icon (string), and a list of departments associated with their job (List<string>)

**Changelog**

:cl:
- tweak: Updated the crew monitoring console user interface

### CODE REVIEW

File: Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs [hunk 1]
```diff
@@ -18,258 +22,309 @@ namespace Content.Client.Medical.CrewMonitoring
...
+                    var index = MathF.Round(4f * (sensor.TotalDamage.Value / 100f));
+
+                    switch (index)
+                    {
+                        case 0: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health0"); break;
+                        case 1: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health1"); break;
+                        case 2: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health2"); break;
+                        case 3: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health3"); break;
+                        case 4: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health4"); break;
+                        default: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "critical"); break;
```
Comment (psykzz) [inline Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs:149]: Is this doing IO each loop?
Do these need to be preloaded beforehand?

Comment (deltanedas) [reply Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs:149]: these are also the same sprite just changes the state

Comment (psykzz) [reply Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs:149]: I guess i need to just check if creating a new ResPath does any IO, but i get your point.

Comment (moonheart08) [reply Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs:149]: tl;dr no

**Comment by Rainbeon** [comment]
This doesn't seem to show at a glance who actually has coordinates on and who just has vitals on, as their location only seems to be revealed once you actually click on them. Would still like to see coordinates put in-line with the other information if possible, or atleast some kind of indicator whether or not they're on before clicking.

**Comment by chromiumboy** [comment]
> This doesn't seem to show at a glance who actually has coordinates on and who just has vitals on, as their location only seems to be revealed once you actually click on them. Would still like to see coordinates put in-line with the other information if possible, or atleast some kind of indicator whether or not they're on before clicking.

If they don't have coords on then the button is disabled and cannot be clicked. It is a little subtle though. I could add a little icon to indicate if coords are on. I didn't want to clutter each line with too many elements though

**Review by metalgearsloth — APPROVED** [review]

File: Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs [hunk 2]
```diff
@@ -18,258 +22,308 @@ namespace Content.Client.Medical.CrewMonitoring
...
-        public CrewMonitoringWindow(EntityUid? mapUid)
+        public CrewMonitoringWindow(string stationName, EntityUid? mapUid)
         {
             RobustXamlLoader.Load(this);
-            _eye = IoCManager.Resolve<IEyeManager>();
+
             _entManager = IoCManager.Resolve<IEntityManager>();
-            _stationUid = mapUid;
+            _prototypeManager = IoCManager.Resolve<IPrototypeManager>();
+            _spriteSystem = _entManager.System<SpriteSystem>();
```
[the remark this answers is not in the document]
Comment (chromiumboy) [reply Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs:38]: Thanks for the tip, I'll look into it

**Comment by chromiumboy** [comment]
Just adding one last features...

**Comment by Rainbeon** [comment]
> If they don't have coords on then the button is disabled and cannot be clicked. It is a little subtle though. I could add a little icon to indicate if coords are on. I didn't want to clutter each line with too many elements though

I know it's already merged but a little icon indicator like we had with the ? and arrows prior would be very very appreciated, could probably omit showing exact coordinates and just have the icon

**Comment by Rainbeon** [comment]
> > > If they don't have coords on then the button is disabled and cannot be clicked. It is a little subtle though. I could add a little icon to indicate if coords are on. I didn't want to clutter each line with too many elements though
> > 
> > 
> > I know it's already merged but a little icon indicator like we had with the ? and arrows prior would be very very appreciated, or even just on/off
> 
> I added a colored dot to the start of each entry, green for on, red for off. You can see them in the image in the media section

Thank you! Very appreciated

Also from testing in live rounds, Crew Monitors no longer seem to reveal the location of armed Nukes anymore? Unsure if an intentional removal or not

**Comment by Brandon-Huu** [comment]
God i hate this change

**Comment by UbaserB** [comment]
Its a good idea, but there are several implementations that i dont like and makes my job harder. One, damage values. Now i have no idea specifically how hurt someone is. Two, the pink hurts my eyes. If everywhere else uses it like station maps for example, why change it?

**Comment by chromiumboy** [comment]
> Its a good idea, but there are several implementations that i dont like and makes my job harder. One, damage values. Now I have no idea specifically how hurt someone is. Two, the pink hurts my eyes. If everywhere else uses it like station maps for example, why change it?

Well, you certainly don't have to know how much damage someone has received to use the console effectively. What does it matter if someone has 50 damage, or 60? All you really need to know is roughly how much damage someone have taken to gauge how how much danger they are in. Plus, AFAIK, the plan is still to phase out health analyzers, so the crew monitoring console will not be listing specific damage values in the long run in any case

The wall color is actually console specific. My idea was to have a different colored walls to differentiate all the consoles that feature a map (the power monitor console will use light blue for example, and I have some ideas for other consoles that a map would be handy). The choice of 'surgical pink' was chosen because it has a nice color contrast with green and cyan, which is the colors used for the blips on the map. I'm not opposed to using another color scheme though, if I find a good suggestion for it

**Comment by UbaserB** [comment]
> > Its a good idea, but there are several implementations that i dont like and makes my job harder. One, damage values. Now I have no idea specifically how hurt someone is. Two, the pink hurts my eyes. If everywhere else uses it like station maps for example, why change it?
> 
> Well, you certainly don't have to know how much damage someone has received to use the console effectively. What does it matter if someone has 50 damage, or 60? All you really need to know is roughly how much damage someone have taken to gauge how how much danger they are in. Plus, AFAIK, the plan is still to phase out health analyzers, so the crew monitoring console will not be listing specific damage values in the long run in any case
> 
> The wall color is actually console specific. My idea was to have a different colored walls to differentiate all the consoles that feature a map (the power monitor console will use light blue for example, and I have some ideas for other consoles that a map would be handy). The choice of 'surgical pink' was chosen because it has a nice color contrast with green and cyan, which is the colors used for the blips on the map. I'm not opposed to using another color scheme though, if I find a good suggestion for it

what about muted orange? I mean, plasmafire is a new orange UI theme afterall so why not

**Comment by chromiumboy** [comment]
> what about muted orange? I mean, plasmafire is a new orange UI theme afterall so why not

Maybe? Orange strikes me more engineering focused. Got a color code / hex code in mind?

**Comment by metalgearsloth** [comment]
If we were changing I would just go with the departmental color which is under the Sixteen colorset iirc.

**Pull request merged**